### PR TITLE
chore(ci): bump Github actions to support node.js 20

### PR DESCRIPTION
**Pull request by Water-Melon**
### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

KAG-5461

### CODE REVIEW

**Comment by Water-Melon** [comment]
all jobs:
https://github.com/Kong/kong/actions/runs/11251077175?pr=13734
https://github.com/Kong/kong/actions/runs/11251077136?pr=13734
https://github.com/Kong/kong/actions/runs/11251077189?pr=13734
https://github.com/Kong/kong/actions/runs/11251077126?pr=13734
https://github.com/Kong/kong/actions/runs/11251077128?pr=13734
https://github.com/Kong/kong/actions/runs/11251076933?pr=13734
https://github.com/Kong/kong/actions/runs/11251077189?pr=13734

File: .github/workflows/backport.yml [hunk 1]
```diff
@@ -37,6 +37,6 @@ jobs:
             }
       - name: add label
         if: steps.backport.outputs.was_successful == 'false'
-        uses: actions-ecosystem/action-add-labels@18f1af5e3544586314bbe15c0273249c770b2daf # v1.1.0
+        uses: Kong/action-add-labels@main
```
Comment (Water-Melon) [inline .github/workflows/backport.yml:40]: Let's wait until actions-ecosystem/action-add-labels supports Node.js 20 before switching back.

File: .github/workflows/cherry-picks.yml [hunk 2]
```diff
@@ -50,6 +50,6 @@ jobs:
             }
       - name: add label
         if: steps.cherry_pick.outputs.was_successful == 'false'
-        uses: actions-ecosystem/action-add-labels@18f1af5e3544586314bbe15c0273249c770b2daf # v1.1.0
+        uses: Kong/action-add-labels@main
```
Comment (Water-Melon) [inline .github/workflows/cherry-picks.yml:53]: Let's wait until actions-ecosystem/action-add-labels supports Node.js 20 before switching back.

File: .github/workflows/backport.yml [hunk 3]
```diff
@@ -37,6 +37,6 @@ jobs:
             }
       - name: add label
         if: steps.backport.outputs.was_successful == 'false'
-        uses: actions-ecosystem/action-add-labels@18f1af5e3544586314bbe15c0273249c770b2daf # v1.1.0
+        uses: Kong/action-add-labels@81b0a07d6b2ec64d770be1ca94c31ec827418054
```
Comment (Water-Melon) [inline .github/workflows/backport.yml:40]: Switch back once `actions-ecosystem/action-add-labels` supports Node.js 20.

File: .github/workflows/cherry-picks.yml [hunk 4]
```diff
@@ -50,6 +50,6 @@ jobs:
             }
       - name: add label
         if: steps.cherry_pick.outputs.was_successful == 'false'
-        uses: actions-ecosystem/action-add-labels@18f1af5e3544586314bbe15c0273249c770b2daf # v1.1.0
+        uses: Kong/action-add-labels@81b0a07d6b2ec64d770be1ca94c31ec827418054
```
Comment (Water-Melon) [inline .github/workflows/cherry-picks.yml:53]: Switch back once `actions-ecosystem/action-add-labels` supports Node.js 20.

**Review by ADD-SP — APPROVED** [review]

**Review by windmgc — APPROVED** [review]

**Pull request merged**